### ISSUE TITLE
More specific feedback on incorrect CSV column headers

### ISSUE THREAD

**Issue by dezog** [data-to-insight/csc-validator-be-903#549]
Currently if a csv file doesn't match precisely one of the accepted headers, the error message just prints the entire header.

An attempt should be made to identify which file it is supposed to be, and if possible highlight which fields need to be renamed/added/removed.

**Comment by CeeJay68** [comment]
Hi Sorry to say it's still not having it, I've double checked for any weird formatting and spaces.
![File Header](https://user-images.githubusercontent.com/89516354/150809150-509db804-54f3-457b-b0cf-9ccd0c47b62f.png)
![Error](https://user-images.githubusercontent.com/89516354/150809700-4ef77cde-2d9c-4398-996c-56c7a7fb7d91.png)

**Comment by dezog** [comment]
I think this is because it's expecting DOB as well:

`CHILD,DOB,DATE_PLACED,DATE_PLACED_CEASED,REASON_PLACED_CEASED`

We've just been going off what's in the tech spec document [here](https://assets.publishing.service.gov.uk/government/uploads/system/uploads/attachment_data/file/924248/Children_looked_after_by_local_authorities_2021-22_Technical_specification_Version_1.0.pdf), I'm not sure if DfE's portal can fill this info in for itself from the Header file if it's not present in other tables

Either way better feedback would definitely be helpful - I was stumped by this for a good while there, and I'm sure other people will run into similar issues!

**Comment by CeeJay68** [comment]
Ah okay thanks, I guess someone this end has made a change to the report and not checked or said anything.

Carl Peirce
***@***.***<https://www.dorsetcouncil.gov.uk/>
Management Information Officer
Business Intelligence & Performance
Dorset Council
01305 224309 <tel:01305%20224309%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20>
dorsetcouncil.gov.uk<https://www.dorsetcouncil.gov.uk>

***@***.***<https://www.facebook.com/DorsetCouncilUK>
***@***.***<https://instagram.com/DorsetCouncilUK>
***@***.***<https://twitter.com/DorsetCouncilUK>

**Comment by PhilDeakin** [comment]
It might be better if we made it say something like   "expected columns ['CHILD','DOB','DATE_PLACED' .... )   but received columns ('CHILD','DATE_PLACED' ... )  ... check the column names in the file you're trying to upload."

**Comment by dezog** [comment]
Agree, that's a great idea - maximum info, minimum coding complexity!

**Comment by CeeJay68** [comment]
Whole heartedly agree

Carl Peirce
***@***.***<https://www.dorsetcouncil.gov.uk/>
Management Information Officer
Business Intelligence & Performance
Dorset Council
01305 224309 <tel:01305%20224309%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20>
dorsetcouncil.gov.uk<https://www.dorsetcouncil.gov.uk>

***@***.***<https://www.facebook.com/DorsetCouncilUK>
***@***.***<https://instagram.com/DorsetCouncilUK>
***@***.***<https://twitter.com/DorsetCouncilUK>
